vault/resource_pki_secret_backend_cert.go

```diff
@@ -172,6 +172,12 @@ func pkiSecretBackendCertResource() *schema.Resource {
 				Default:     false,
 				Description: "Revoke the certificate upon resource destruction.",
 			},
+			consts.FieldRevokeWithKey: {
```

Comment:
I think it would prevent user confusion if `revoke` and `revoke_with_key` were keys that conflicted with each other and not that you needed to set both to get the revoke_with_key behavior. 

We could use the `ConflictsWith` argument to the schema.Resource to enforce this and update the `pkiSecretBackendCertDelete` function to trigger if either is true?